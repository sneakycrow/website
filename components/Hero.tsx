import Image from "next/image";
import React from "react";
import Link from "next/link";
import { default as cx } from "classnames";

interface HeroProps {
  imageURL?: string;
  title?: string;
  subtitle?: string;
  className?: string;
}

const Hero = (props: HeroProps) => {
  const {
    imageURL = "/logo_v2.svg",
    title = "Sneaky Crow",
    subtitle = "software wizard ",
  } = props;

  const wrapperClass = cx(
    "p-6 bg-white grid grid-flow-row md:grid-flow-col justify-center",
    props.className
  );
  return (
    <div className={wrapperClass}>
      <Link
        href="/"
        className="inline-flex justify-self-center md:justify-self-end min-w-3/4"
      >
        <Image
          alt=""
          src={imageURL}
          width={200}
          height={200}
          className="min-w-[200px]"
        />
      </Link>
      <div className="flex flex-col justify-center items-center font-extrabold">
        <h1 className="w-full text-lg lg:text-3xl text-center lg:text-left">
          {title}
        </h1>
        <h2 className="w-full text-base lg:text-2xl leading-13 text-center lg:text-left">
          {subtitle}
        </h2>
      </div>
    </div>
  );
};

export default Hero;
