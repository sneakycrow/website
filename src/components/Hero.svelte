<script lang="ts">
  export let homeURL: string;
  export let imageURL: string;
  export let title: string;
  export let subtitle: string;
</script>

<div class="p-6 bg-white grid grid-flow-row md:grid-flow-col justify-center">
  <a
    class="inline-flex justify-self-center md:justify-self-end min-w-3/4"
    href={homeURL}
  >
    <img
      alt="Line art picture of a crow with a green baseball cap on, logo of Sneaky Crow"
      class="min-w-[200px]"
      height={200}
      src={imageURL}
      width={200}
    />
  </a>
  <div class="flex flex-col justify-center items-center font-extrabold">
    <h1 class="w-full text-lg lg:text-3xl text-center lg:text-left">
      {title}
    </h1>
    <h2 class="w-full text-2xl leading-13 text-center lg:text-left">
      {subtitle}
    </h2>
  </div>
</div>