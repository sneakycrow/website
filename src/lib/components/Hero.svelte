<script lang="ts">
  export let homeURL: string = "/";
  export let title: string;
  export let subtitle: string;
</script>

<div class="bg-white grid grid-flow-row md:grid-flow-col justify-center">
  <a class="inline-flex justify-self-center md:justify-self-end min-w-3/4" href={homeURL}>
    <img
      alt="Line art of a crow with a green baseball cap on, logo of Sneaky Crow"
      class="min-w-[200px]"
      height={200}
      src="/logo_v2.svg"
      width={200}
    />
  </a>
  <div class="flex flex-col justify-center items-center font-extrabold">
    <h1 class="w-full text-lg lg:text-2xl text-center lg:text-left">
      {title}
    </h1>
    <h2 class="w-full text-xl leading-13 text-center lg:text-left">
      {subtitle}
    </h2>
  </div>
</div>
