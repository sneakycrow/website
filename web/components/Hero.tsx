import Image from "next/image";
import React from "react";
import Link from "next/link";

interface HeroProps {
  imageURL?: string;
  title?: string;
  subtitle?: string;
}

const Hero = (props: HeroProps) => {
  const {
    imageURL = "/logo_v2.svg",
    title = "Sneaky Crow",
    subtitle = "software wizard ",
  } = props;
  return (
    <div className="p-6 bg-white grid grid-flow-row md:grid-flow-col justify-center">
      <Link
        href="/"
        className="inline-flex justify-self-center md:justify-self-end min-w-3/4"
      >
        <Image
          alt=""
          src={imageURL}
          width={200}
          height={200}
          className="min-w-[200px]"
        />
      </Link>
      <div className="flex flex-col justify-center items-center font-extrabold">
        <h1 className="text-3xl text-center lg:text-left">{title}</h1>
        <h2 className="text-2xl leading-13 text-center lg:text-left">
          {subtitle}
        </h2>
      </div>
    </div>
  );
};

export default Hero;
