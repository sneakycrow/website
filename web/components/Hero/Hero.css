.title {
    position: relative;
    z-index: 1;
    line-height: 5rem;
    font-size: 4.8rem;
}

.title:hover::after {
    width: 100%;
}

.title::after {
    content: "";
    background-color: #0ba750;
    width: 25%;
    height: 4px;
    position: absolute;
    left: 0;
    bottom: 0;
    z-index: -1;
    transition: width 250ms ease-in-out;
}