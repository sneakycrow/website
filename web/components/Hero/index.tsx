import Image from "next/image";
import React from "react";
import Link from "next/link";
import "./Hero.css";

interface HeroProps {
  imageURL?: string;
  title?: string;
  subtitle?: string;
}

const Index = (props: HeroProps) => {
  const {
    imageURL = "/logo_v2.svg",
    title = "Sneaky Crow",
    subtitle = "software wizard ",
  } = props;
  return (
    <div className="p-6 bg-white grid grid-flow-row lg:grid-flow-col">
      <Link href="/" className="inline-flex justify-center min-w-3/4">
        <Image alt="" src={imageURL} width={200} height={200} />
      </Link>
      <div className="flex flex-col justify-center items-center lg:items-start font-extrabold">
        <h1 className="title">{title}</h1>
        <h2 className="text-2xl leading-13">{subtitle}</h2>
      </div>
    </div>
  );
};

export default Index;
