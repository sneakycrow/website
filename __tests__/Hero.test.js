import React from 'react';
import { render } from '@testing-library/react';
import Hero from '../components/hero';
import ThemeToggler from '../components/themeToggler';

describe('<Hero /> spec', () => {
  it('renders the component', () => {
    const container = render(
      <ThemeToggler>
        <Hero>
          <h1>Testing</h1>
        </Hero>
      </ThemeToggler>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
