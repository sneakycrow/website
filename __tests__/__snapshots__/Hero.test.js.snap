// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> spec renders the component 1`] = `undefined`;
